endpoints/openrtb2/amp_auction.go

```diff
@@ -348,7 +348,7 @@ func (deps *endpointDeps) loadRequestJSONForAmp(httpRequest *http.Request) (req
 		return
 	}
 
-	if deps.cfg.GenerateRequestID {
+	if deps.cfg.GenerateRequestID || req.ID == "{{UUID}}" {
```

Comment:
Is `req.ID` referencing the stored request, or the original request? The token should only apply to when it's used in the stored request.

Reply:
In AMP, there is no original request. So it would have to be the stored request. (Or possibly the query string)